test/transformers/test_rms_norm.py

```diff
@@ -24,6 +24,22 @@ def forward(self, hidden_states):
         return self.weight * hidden_states.to(input_dtype)
 
 
+# https://github.com/huggingface/transformers/blob/v4.44.2/src/transformers/models/gemma/modeling_gemma.py#L122
+class GemmaRMSNorm(nn.Module):
+    def __init__(self, hidden_size: int, eps: float = 1e-6):
+        super().__init__()
+        self.eps = eps
+        self.weight = nn.Parameter(torch.ones(hidden_size))
+
+    def _norm(self, x):
+        return x * torch.rsqrt(x.pow(2).mean(-1, keepdim=True) + self.eps)
+
+    def forward(self, x):
+        output = self._norm(x.float())
```

Comment:
@yundai424 I am curious if we need to do the upcasting too in our kernel? or we don't because anyway it gets casted to x.dtype. Our old impl doesn't have that but the result is still consistent

Reply:
I agree that it might not be the exact same, but in the tests, the results are consistent with Gemma too (up to the testing tolerance). I think it's a tradeoff between exact reproduction and complexity (because I suppose different models might have different small details like this casting thing (llama casting != gemma casting), so handling every variation might end up with a lot of code in the future.

Reply:
> @yundai424 I am curious if we need to do the upcasting too in our kernel? or we don't because anyway it gets casted to x.dtype. Our old impl doesn't have that but the result is still consistent

The test results seem to be consistent with the reference too. I think it's a tradeoff between complexity (because some models have slight inconsistencies) and exact reproduction. Let me know what you think and I can modify it (or even on another PR, to keep things clean)

Reply:
lgtm! waiting @yundai424 to final check

Reply:
can we add gemma to convergence test as well?

Reply:
> can we add gemma to convergence test as well?

Sure. Do you think it's better to do it in another PR?

Reply:
> can we add gemma to convergence test as well?

adding it

Reply:
> can we add gemma to convergence test as well?

done. as mentioned in the discord channel, I made the tolerance for bfloat16 convergence test a bit bigger, since the casting difference seems to be slightly affecting it (for the regular tests it's fine, but it seems to add up in the end-to-end training situation)